tests/tests_pytorch/deprecated_api/test_no_removal_version.py

```diff
@@ -53,5 +54,74 @@ def test_double_precision_wrapper():
 
 
 def test_fsdp_mixed_precision_plugin():
+    from lightning.pytorch.plugins.precision.fsdp import FSDPMixedPrecisionPlugin
+
     with pytest.deprecated_call(match=r"The `FSDPMixedPrecisionPlugin` is deprecated"):
         FSDPMixedPrecisionPlugin(precision="16-mixed", device="cuda")
+
+
+def test_fsdp_precision_plugin():
+    from lightning.pytorch.plugins.precision.fsdp import FSDPPrecisionPlugin
+
+    with pytest.deprecated_call(match=r"The `FSDPPrecisionPlugin` is deprecated"):
+        FSDPPrecisionPlugin(precision="16-mixed")
+
+
+@pytest.mark.skipif(not _BITSANDBYTES_AVAILABLE, reason="bitsandbytes unavailable")
+def test_bitsandbytes_precision_plugin():
+    from lightning.pytorch.plugins.precision.bitsandbytes import BitsandbytesPrecisionPlugin
+
+    with pytest.deprecated_call(match=r"The `BitsandbytesPrecisionPlugin` is deprecated"):
+        BitsandbytesPrecisionPlugin("nf4")
+
+
+@RunIf(deepspeed=True)
+def test_deepspeed_precision_plugin():
+    from lightning.pytorch.plugins.precision.deepspeed import DeepSpeedPrecisionPlugin
+
+    with pytest.deprecated_call(match=r"The `DeepSpeedPrecisionPlugin` is deprecated"):
+        DeepSpeedPrecisionPlugin(precision="32-true")
+
+
+def test_double_precision_plugin():
+    from lightning.pytorch.plugins.precision.double import DoublePrecisionPlugin
+
+    with pytest.deprecated_call(match=r"The `DoublePrecisionPlugin` is deprecated"):
+        DoublePrecisionPlugin()
+
+
+def test_half_precision_plugin():
+    from lightning.pytorch.plugins.precision.half import HalfPrecisionPlugin
+
+    with pytest.deprecated_call(match=r"The `HalfPrecisionPlugin` is deprecated"):
+        HalfPrecisionPlugin()
+
+
+def test_mixed_precision_plugin():
+    from lightning.pytorch.plugins.precision.amp import MixedPrecisionPlugin
+
+    with pytest.deprecated_call(match=r"The `MixedPrecisionPlugin` is deprecated"):
+        MixedPrecisionPlugin(precision="16-mixed", device="cuda")
+
+
+def test_precision_plugin():
+    from lightning.pytorch.plugins.precision.precision import PrecisionPlugin
+
+    with pytest.deprecated_call(match=r"The `PrecisionPlugin` is deprecated"):
+        PrecisionPlugin()
+
+
+@pytest.mark.skipif(not _TRANSFORMER_ENGINE_AVAILABLE, reason="transformer-engine unavailable")
+def test_transformer_engine_precision_plugin():
+    from lightning.pytorch.plugins.precision.transformer_engine import TransformerEnginePrecisionPlugin
+
+    with pytest.deprecated_call(match=r"The `TransformerEnginePrecisionPlugin` is deprecated"):
+        TransformerEnginePrecisionPlugin()
+
+
+@RunIf(tpu=True)
```

Comment:
Can you use the `xla_available` fixture instead to mock required dependencies? This test doesn't need to run on a TPU device.

I suggest doing the same for the other deprecated plugins. For instance, TransformerEngine and Bitsandbytes are not installed at all in CI. You can follow the mocking pattern in their existing tests